File: game-logger/Domain/Models/DTOs/PlayerActionDto.cs
using System;
using Domain.Enums;

namespace Domain.Models.DTOs;

public class PlayerActionDto
{
    public Guid TargetNodeId { get; set; }
    public int NumberOfUnits { get; set; }
    public int TickActionCompleted { get; set; }
    public int TickActionStart { get; set; }
    public int TickReceived { get; set; }
    public ActionType ActionType { get; set; }
}

File: starter-bots/KotlinBot/src/main/kotlin/arctica/service/model/dto/PlayerActionDto.kt
package arctica.service.model.dto

import arctica.service.model.enums.ActionType
import java.util.UUID

data class PlayerActionDto(

    val targetNodeId: UUID,
    val numberOfUnits: Int,
    val tickActionCompleted: Int,
    val tickActionStart: Int,
    val tickReceived: Int,
    val actionType: ActionType
)